node/src/components/storage.rs

```diff
@@ -1756,6 +1756,16 @@ impl Storage {
         ret
     }
 
+    /// Retrieves single switch block header by era ID by looking it up in the index and returning
+    /// it.
+    pub(crate) fn read_switch_block_header_by_era_id(
```

Comment:
It's not directly related to this PR, but since we now have this function, we might double check if `get_switch_block_by_era_id()` is still needed.

It's plausible that `get_switch_block_by_era_id()` has been used to only access data from the header, so now we could potentially short-circuit this by leveraging this new function.